core/dbt/task/list.py

```diff
@@ -148,9 +139,14 @@ def run(self):
         return self.output_results(generator())
 
     def output_results(self, results):
+        """Log, or output a plain, newline-delimited, and ready-to-pipe list of nodes found."""
         for result in results:
             self.node_results.append(result)
-            print(result)
+            if get_flag_dict()["log_format"] == "json":
+                fire_event(ListRunDetails(msg=result))
+            else:
+                # Cleaner to leave as print than to mutate the logger.
+                print(result)
```

Comment:
This feels like the right idea. I'm not sure if it's the exact implementation we'll want, versus swapping this out for a more generic `clients.system.print` or `events.functions.print` method that handles this conditional logic for us.

I opened an issue to encapsulate the larger need here: https://github.com/dbt-labs/dbt-core/issues/6543

Reply:
Normally I would expect to see somebody use flags.LOG_FORMAT to get the flag setting. Though I'm not sure what's happening with the flags object in the about-to-me merged click branch.

Also I'm not sure what the reference to having to mutate the logger is about. What issue were you avoiding here?

Reply:
re: the flags question.

For sure! I tried that aaaaaand it doesn't work here. 😕 For whatever reason, under certain combinations of passing in the `json` type, the LOG_FORMAT object is None. Maybe that's a deeper bug with `flags.py`?

Reply:
Using "get_flag_dict" should be no different, since it just copies the value of LOG_FORMAT into the dictionary. Use flags.LOG_FORMAT and if you get a None we should figure out why.

Reply:
I'll for sure double check this, but at the time I did this work, the values I was getting were different for whatever reason. It could have been due to logger mutations that no longer exist or something else. Rest assured, I'll be sure to asses.

Reply:
Oh good news, some upstream change on merge fixed this for me. I'll simplify as you suggest 👏